sbi/inference/snpe/snpe_a.py

```diff
@@ -1,154 +1,34 @@
 from __future__ import annotations
-
-import os
+from typing import Callable, Optional
 
 import torch
-from torch import distributions
+from torch import Tensor, nn
 from torch.utils.tensorboard import SummaryWriter
 
-import sbi.utils as utils
 from sbi.inference.snpe.snpe_base import SnpeBase
 
 
 class SnpeA(SnpeBase):
     def __init__(
         self,
-        simulator,
+        simulator: Callable,
         prior,
-        x_o,
-        num_pilot_sims=100,
-        density_estimator="maf",
-        use_combined_loss=False,
-        z_score_x=True,
-        simulation_batch_size: int = 1,
-        retrain_from_scratch_each_round=False,
-        discard_prior_samples=False,
-        summary_writer=None,
-        device=None,
+        x_o: Tensor,
+        density_estimator: Optional[nn.Module],
```

Comment:
Is `Optional[nn.Module]` enough to enable to user pass nothing? Shouldnt it be `Optional[nn.Module] = None`?

Reply:
You can pass either `None` or an `nn.Module`. The type doesn't force you to set a default value. Are there reasons to set it in this particular case?
EDIT: it is default `None` in `Snpe(Base|B|C)`, just not in `A`.